src/Compilers/CSharp/Portable/Symbols/ExtraAnnotations.cs

```diff
@@ -0,0 +1,201 @@
+                { "System.Byte System.Buffer.GetByte(System.Array, System.Int32)", Array(default, Nullable(false), default) },
+                { "System.Void System.Buffer.SetByte(System.Array, System.Int32, System.Byte)", Array(default, Nullable(false), default, default) },
+                { "System.Byte System.Byte.Parse(System.String)", Array(default, Nullable(false)) },
+                { "System.Byte System.Byte.Parse(System.String, System.IFormatProvider)", Array(default, Nullable(false), default) },
+                { "System.Byte System.Byte.Parse(System.String, System.Globalization.NumberStyles)", Array(default, Nullable(false), Nullable(false)) },
+                { "System.Byte System.Byte.Parse(System.String, System.Globalization.NumberStyles, System.IFormatProvider)", Array(default, Nullable(false), Nullable(false), default) },
+                { "System.String System.String.Concat(System.String, System.String)", Array(Nullable(false), Nullable(true), Nullable(true)) },
+            }.ToImmutableDictionary();
+
+        private static readonly ImmutableDictionary<string, ImmutableArray<AttributeAnnotations>> Attributes =
+            new Dictionary<string, ImmutableArray<AttributeAnnotations>>
+            {
+                { "System.Boolean System.String.IsNullOrEmpty(System.String)", Array(default, NotNullWhenFalse) },
+                { "System.Boolean System.String.IsNullOrWhiteSpace(System.String)", Array(default, NotNullWhenFalse) },
+                { "System.Boolean System.String.Contains(System.String)", Array(default, EnsuresNotNull) },
+            }.ToImmutableDictionary();
+
+        internal static string MakeMethodKey(PEMethodSymbol method, ParamInfo<TypeSymbol>[] paramInfo)
+        {
+            var pooledBuilder = PooledStringBuilder.GetInstance();
+
+            StringBuilder builder = pooledBuilder.Builder;
+            Add(paramInfo[0].Type, builder);
+            builder.Append(' ');
+
+            Add(method.ContainingType, builder);
+            builder.Append('.');
+
+            builder.Append(method.Name);
+            builder.Append('(');
+
+            for (int i = 1; i < paramInfo.Length; i++)
+            {
+                Add(paramInfo[i].Type, builder);
+                if (i < paramInfo.Length - 1)
+                {
+                    builder.Append(", ");
+                }
+            }
+
+            builder.Append(')');
+            return pooledBuilder.ToStringAndFree();
+
+            // PROTOTYPE(NullableReferenceTypes): Many cases are not yet handled
+            // generic type args
+            // ref kind
+            // 'this'
+            // static vs. instance
+            // use assembly qualified name format (used in metadata) rather than symbol display?
+        }
+
+        internal static string MakeMethodKey(MethodSymbol method)
+        {
+            var containingType = method.ContainingSymbol as TypeSymbol;
+            if (containingType is null)
```

Comment:
When is `method.ContainingSymbol` not a `TypeSymbol`?

Reply:
I think for local functions in can be a method symbol.

---
In reply to: [187470794](https://github.com/dotnet/roslyn/pull/26656#discussion_r187470794) [](ancestors = 187470794)